toolkit/components/uniffi-bindgen-gecko-js/src/interface_ext.rs

```diff
@@ -222,3 +226,118 @@ pub impl FFIArgument {
         matches!(self.type_(), FFIType::RustBuffer)
     }
 }
+
+
+#[ext]
+pub impl Record {
+    fn js_name(&self) -> String {
+        self.name().to_camel_case()
+    }
+
+    fn js_constructor_field_list(&self) -> String {
+        let mut field_list = String::new();
+        for (idx, field) in self.fields().iter().enumerate() {
+            let field = field.name().to_mixed_case();
+            field_list.push_str(&field);
+            if idx != self.fields().len() - 1 {
+                field_list.push_str(",")
+            }
```

Comment:
I like having the code here rather than in a askama macro like the other bindings.  But I think it could be simplified more by using the `Vec::join()` method.  I think you could just do `self.fields().join(",")`.